eqc/tcp_proto_eqc.erl

```diff
@@ -244,3 +247,42 @@ prop_encode_decode_metrics() ->
                 L2 = lists:sort(Rev),
                 L1 == L2
             end).
+
+prop_encode_decode_bucket_info() ->
+    ?FORALL(Msg = {Resolution, _, _}, {resolution(), pos_int(), ttl()},
+            case valid_time(Resolution) of
+                true ->
+                    Encoded = dproto_tcp:encode_bucket_info(Msg),
+                    Decoded = dproto_tcp:decode_bucket_info(Encoded),
+                    ?WHENFAIL(
+                       io:format(user,
+                                 "~p -> ~p -> ~p~n",
+                                 [Msg, Encoded, Decoded]),
+                       Msg =:= Decoded);
+                _ ->
+                    {'EXIT', _} = (catch dproto_tcp:encode(Msg))
+            end).
+
+prop_encode_decode_ttl() ->
+    ?FORALL(Msg = {ttl, Bucket, TTL}, {ttl, bucket(), ttl()},
+            case TTL of
+                infinity ->
+                    Encoded = dproto_tcp:encode(Msg),
+                    Decoded = dproto_tcp:decode(Encoded),
+                    ?WHENFAIL(
+                       io:format(user,
+                                 "~p -> ~p -> ~p~n",
+                                 [Msg, Encoded, Decoded]),
+                       {ttl, Bucket, 0} =:= Decoded);
```

Comment:
you can collapse the two case statements into 1 with an `; TTL =:= infinity` I think